docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_webhook.go

```diff
@@ -118,40 +160,80 @@ validate anything on deletion.
 // NOTE: The 'path' attribute must follow a specific pattern and should not be modified directly here.
 // Modifying the path for an invalid path can cause API server errors; failing to locate the webhook.
 
-var _ webhook.Validator = &CronJob{}
+// +kubebuilder:object:generate=false
+// CronJobCustomValidator struct is responsible for validating the CronJob resource
+// when it is created, updated, or deleted.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as this struct is used only for temporary operations and does not need to be deeply copied.
+type CronJobCustomValidator struct {
+	cronjob *CronJob
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *CronJob) ValidateCreate() (admission.Warnings, error) {
-	cronjoblog.Info("validate create", "name", r.Name)
+	//TODO(user): Add more fields as needed for validation
+}
+
+var _ webhook.CustomValidator = &CronJobCustomValidator{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type CronJob
+func (v *CronJobCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	cronjoblog.Info("Creation Validation for CronJob")
+	cronjob, ok := obj.(*CronJob)
+	if !ok {
+		return nil, fmt.Errorf("expected a CronJob object but got %T", obj)
+	}
+	cronjoblog.Info("default", "name", cronjob.GetName())
```

Comment:
Not "default" here, but same as the log above. This will produce a huge amount of noise in the controller logs (also the log in l.179)

Reply:
Good point about the default
we should get the request namespace 
But I think we need to keep the logs. I will give a look with love thx

Reply:
Up to you. But I think in a regular use cases defaulting and validating webhooks are called a lot. Logging this with log level 0 will lead to a lot of noise